cgen: error when using `or { }` block in struct field init

As title states

```v
struct RegexCache {
mut:
    tag_script_start    regex.RE = regex.regex_opt(r'^script.*>') or { panic(err) } // <- This gives some trouble in some cases
}
```

`test.v`
```v
import regex

struct RegexCache {
mut:
    tag_script_start    regex.RE = regex.regex_opt(r'^script.*>') or { panic(err) }
}

pub struct Tokenizer {
mut:
    regex_cache RegexCache = RegexCache{}
}

fn new_parser() &Parser {
    mut parser := &Parser{}
    return parser
}

pub struct Parser {
mut:
    tnzr        Tokenizer
}

fn main() {
    p := new_parser()
    println(p)
}
```

`v run test.v`
```
/dev/shm/v/run.7054598677465549469.tmp.c:12749: warning: implicit declaration of function '_t'
/dev/shm/v/run.7054598677465549469.tmp.c:12749: error: ')' expected (got "{")
```

C code around the issue:
```c
VV_LOCAL_SYMBOL main__Parser* main__new_parser(void) {
	Option_regex__RE _t1 = regex__regex_opt(_SLIT("^script.*>"));
	if (_t(main__Tokenizer){.regex_cache = 1.state != 0) { /*or block*/ 
		IError err = _t1.err;
		_v_panic(IError_str(err));
		VUNREACHABLE();
	;
	}
	
 	main__Parser* parser = (main__Parser*)memdup(&(main__Parser){.tnzr = (main__RegexCache){.tag_script_start =  (*(regex__RE*)_t1.data),},},}, sizeof(main__Parser));
	main__Parser* _t2 = parser;
	return _t2;
}
```
![image](https://user-images.githubusercontent.com/768942/128756673-2611ed45-ad56-4070-a43e-f10a40f044fa.png)

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -415,6 +415,14 @@ fn (mut g Gen) zero_struct_field(field ast.StructField) bool {
 					g.struct_init(default_init)
 				}
 				return true
+			} else if sym.language == .v && !field.typ.is_ptr() && sym.mod != 'builtin'
+				&& !sym.info.is_empty_struct() {
+				default_init := ast.StructInit{
+					typ: field.typ
+				}
+				g.write('.${field_name} = ')
+				g.struct_init(default_init)
+				return true
 			}
 		}
 	}

--- a/vlib/v/tests/structs/struct_field_default_value_optional_test.v
+++ b/vlib/v/tests/structs/struct_field_default_value_optional_test.v
@@ -1,0 +1,27 @@
+import regex
+
+struct RegexCache {
+mut:
+	tag_script_start regex.RE = regex.regex_opt(r'^script.*>') or { panic(err) }
+}
+
+pub struct Tokenizer {
+mut:
+	regex_cache RegexCache = RegexCache{}
+}
+
+fn new_parser() &Parser {
+	mut parser := &Parser{}
+	return parser
+}
+
+pub struct Parser {
+mut:
+	tnzr Tokenizer
+}
+
+fn test_struct_field_default_value_optional() {
+	p := new_parser()
+	println(p)
+	assert true
+}

